video_out_sdl.c

```diff
@@ -0,0 +1,411 @@
+          instance->bpp, instance->sdlflags);
+    SDL_DisplayYUVOverlay (overlay, &(instance->surface->clip_rect));
+}
+
+static void sdl_discard (vo_instance_t * _instance,
+       uint8_t * const * buf, void * id)
+{
+    SDL_UnlockYUVOverlay ((SDL_Overlay *) id);
+}
+
+static void sdl_close (vo_instance_t * _instance)
+{
+    sdl_instance_t * instance;
+    int i;
+
+    instance = (sdl_instance_t *) _instance;
+    for (i = 0; i < 3; i++)
+  SDL_FreeYUVOverlay (instance->frame[i].overlay);
+    SDL_FreeSurface (instance->surface);
+    SDL_QuitSubSystem (SDL_INIT_VIDEO);
+}
+#endif
+
+static int sdl_setup (vo_instance_t * _instance, unsigned int width,
+          unsigned int height, unsigned int chroma_width,
+          unsigned int chroma_height, vo_setup_result_t * result)
+{
+    sdl_instance_t * instance = (sdl_instance_t *) _instance;
+
+    instance->width = width;
+    instance->height = height;
+    instance->surface = SDL_SetVideoMode (width, height, instance->bpp,
+            instance->sdlflags);
+    if (! (instance->surface)) {
+  fprintf (stderr, "sdl could not set the desired video mode\n");
+  return 1;
+    }
+    instance->rgb = SDL_CreateRGBSurface(SDL_SRCCOLORKEY, width, height, 24, 0x0000FF, 0x00FF00, 0xFF0000, 0);
+    if (! (instance->rgb)) {
+  fprintf (stderr, "sdl could not create RGB surface\n");
+  return 1;
+    }
+
+    //result->convert = NULL;
+    return 0;
+}
+
+vo_instance_t * vo_sdl_open (void)
+{
+    sdl_instance_t * instance;
+    const SDL_VideoInfo * vidInfo;
+
+    instance = (sdl_instance_t *) malloc (sizeof (sdl_instance_t));
+    if (instance == NULL)
+  return NULL;
+
+    memset (instance, 0, sizeof (sdl_instance_t));
+    //instance->vo.setup = sdl_setup;
+    //instance->vo.setup_fbuf = sdl_setup_fbuf;
+    //instance->vo.set_fbuf = NULL;
+    //instance->vo.start_fbuf = sdl_start_fbuf;
+    //instance->vo.discard = sdl_discard;
+    //instance->vo.draw = sdl_draw_frame;
+    //instance->vo.close = NULL; /* sdl_close; */
+    instance->sdlflags = SDL_HWSURFACE | SDL_ASYNCBLIT | SDL_HWACCEL;
+
+    //putenv("SDL_VIDEO_YUV_HWACCEL=1");
+    //putenv("SDL_VIDEO_X11_NODIRECTCOLOR=1");
+
+    if (SDL_Init (SDL_INIT_VIDEO)) {
+  fprintf (stderr, "sdl video initialization failed.\n");
+  return NULL;
+    }
+
+    vidInfo = SDL_GetVideoInfo ();
+    if (!SDL_ListModes (vidInfo->vfmt, SDL_HWSURFACE | SDL_RESIZABLE)) {
+  instance->sdlflags |= SDL_RESIZABLE;
+  if (!SDL_ListModes (vidInfo->vfmt, SDL_RESIZABLE)) {
+      fprintf (stderr, "sdl couldn't get any acceptable video mode\n");
+      return NULL;
+  }
+    }
+    instance->bpp = vidInfo->vfmt->BitsPerPixel;
+    if (instance->bpp < 16) {
+  fprintf(stderr, "sdl has to emulate a 16 bit surfaces, "
+    "that will slow things down.\n");
+  instance->bpp = 16;
+    }
+
+    return (vo_instance_t *) instance;
+}
+#endif
+
+void ShowHelp(char **ta)
+{
+  char *t;
+  while ((t = *ta)) {
+    printf("%s\n", t);
```

Comment:
See for example http://stackoverflow.com/questions/22886500/how-to-render-text-in-sdl2